### ISSUE TITLE
[REVIEW]: limHaloPT: A numerical Package for Accurate Modelling of Line Intensity Power spectrum

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4173]
**Submitting author:** @amoradinejad (<a href="http://orcid.org/0000-0001-8841-9989">Azadeh Moradinezhad Dizgah</a>)
**Repository:** <a href="https://github.com/amoradinejad/limHaloPT" target ="_blank">https://github.com/amoradinejad/limHaloPT</a>
**Version:** v1.0.0
**Editor:** @eloisabentivegna
**Reviewer:** @ziotom78, @abhimaniyar
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/dde5dbbc5317090aa593124dec601c08/status.svg)](https://joss.theoj.org/papers/dde5dbbc5317090aa593124dec601c08)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/dde5dbbc5317090aa593124dec601c08"><img src="https://joss.theoj.org/papers/dde5dbbc5317090aa593124dec601c08/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/dde5dbbc5317090aa593124dec601c08/status.svg)](https://joss.theoj.org/papers/dde5dbbc5317090aa593124dec601c08)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/amoradinejad/limHaloPT" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@ziotom78 & @abhimaniyar, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @eloisabentivegna know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @ziotom78
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/amoradinejad/limHaloPT">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/amoradinejad/limHaloPT">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@amoradinejad) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @abhimaniyar
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/amoradinejad/limHaloPT">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/amoradinejad/limHaloPT">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@amoradinejad) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @ziotom78, @abhimaniyar it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Failed to discover a `Statement of need` section in paper

**Comment by whedon** [comment]
Wordcount for `paper.md` is 816

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1088/1475-7516/2016/07/028 is OK
- 10.1088/1475-7516/2015/02/013 is OK
- 10.1088/1475-7516/2018/05/019 is OK
- 10.1016/j.ascom.2015.05.005 is OK
- 10.1088/1475-7516/2011/07/034 is OK
- 10.3847/1538-3881/aadae0 is OK
- 10.1088/1475-7516/2021/05/067 is OK
- 10.1046/j.1365-8711.2000.03715.x is OK
- 10.1016/S0370-1573(02)00276-4 is OK
- 10.1088/1475-7516/2012/07/051 is OK
- 10.1007/JHEP09(2012)082 is OK
- 10.1103/PhysRevD.96.083528 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04173/joss.04173/10.21105.joss.04173.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04173/joss.04173/10.21105.joss.04173.pdf) :page_facing_up: :point_left:

**Comment by eloisabentivegna** [comment]
@ziotom78. @abhimaniyar, thanks for the progress made so far! Please let me know if you need any assistance.

**Comment by ziotom78** [comment]
@amoradinejad, I have just posted another issue on your repository ([#11](https://github.com/amoradinejad/limHaloPT/issues/11)) to add examples to the README.

I have a general comment. The lack of examples in the README misses one of the requirements for a paper to be accepted by JOSS, as shown in the checklist that is shown above in this page and stated here («Documentation» / «Example usage»):

https://joss.readthedocs.io/en/latest/review_checklist.html

I encourage you to read carefully this list of points. After a quick check, I fear that other points are not addressed (are there «automatic tests»? «community guidelines»?).

If you could have a check of the kind of stuff JOSS reviewers ask, the review process would probably speed up enormously.

**Comment by eloisabentivegna** [comment]
Hi @abhimaniyar, please open a new issue with the requests that weren't answered, and document all of this here (if you link this review in the issue, it will be mentioned here -- see above for an example). In particular, please explain why you are not satisfied with the way your original issue was handled. That way it's easy to reconstruct what exactly has happened.

**Comment by abhimaniyar** [comment]
@amoradinejad <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_amoradinejad&d=DwMFaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=j7NTX35lzotSaEysgWTW31rDj0okOldBCLczNlYAI8w&m=87YU7psLWOQYeS5uKAEh4WOPw7Hvp2FXY9wC5FBHIa8Jh6j1Di48E5mPw3j-IrC2&s=MZ6DcO_fSJvwCDNGYqOgnqusAEaSg6brS2kUom3co80&e=>, I have just posted an issue (#12 <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_amoradinejad_limHaloPT_issues_11&d=DwMFaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=j7NTX35lzotSaEysgWTW31rDj0okOldBCLczNlYAI8w&m=87YU7psLWOQYeS5uKAEh4WOPw7Hvp2FXY9wC5FBHIa8Jh6j1Di48E5mPw3j-IrC2&s=JjPOIZE1Lm8mKFS2Hn10jdTkOpKY73bCu3RvCgGBNf0&e=>) following up on my previous issue (#10 <https://github.com/amoradinejad/limHaloPT/issues/10#issue-1156019803>) to expand your documentation to include more clarity on the sample examples for the users. Kindly update the README accordingly.
Thank you.

**Comment by eloisabentivegna** [comment]
@abhimaniyar, @ziotom78, are there updates regarding this review? Please let us know if you're experiencing any road blocks.

**Comment by amoradinejad** [comment]
@eloisabentivegna <https://github.com/eloisabentivegna> ,  @ziotom78 <https://github.com/ziotom78>, @abhimaniyar
 <https://github.com/abhimaniyar>
To address the two main comments by the referee regarding having tests and clear description of the usage of eta code, I am currently re-writing parts of my code and adding new components. These changes were not just cosmetic, but required some restructuring. Therefore, they have taken some time to be completed. I should be able to fully address the referee’s points shortly. I will post an update. 


Thank you,
Azadeh

**Comment by eloisabentivegna** [comment]
Hi @amoradinejad, do you have an estimate of when the restructuring will be complete?

@ziotom78, @abhimaniyar, is this currently a blocker for you?

**Comment by amoradinejad** [comment]
@eloisabentivegna <https://github.com/eloisabentivegna> , the re-structuring of the code is completed, and I have addressed the comments made by the two referees. So on my side, all is done.

**Comment by ziotom78** [comment]
Sorry for the delay, I have been away for Easter holidays. I have checked the code, and the latest work by @amoradinejad have improved the installation and functionality a lot: great job!

My review is now complete, and I checked every item in the review checklist.

**Comment by eloisabentivegna** [comment]
Thanks for thorough job, @ziotom78!

@abhimaniyar, can you comment on the recent changes as well, and provide us with an update on your review?

**Comment by abhimaniyar** [comment]
@eloisabentivegna I am so sorry I forgot to reply to this! In fact, I had checked every item on the checklist and was done with my review more than 2 week ago! I thought completing the checklist meant completion of the review, I am sorry I should have confirmed that with you. However, I am now very happy to confirm that I am too done with my review. Thanks!

**Comment by eloisabentivegna** [comment]
Thanks @abhimaniyar, and congratulations to the authors for a successful review.

Let's start with the prepublication process.

**Comment by eloisabentivegna** [comment]
@editorialbot generate pdf

**Comment by eloisabentivegna** [comment]
@editorialbot generate pdf

**Comment by eloisabentivegna** [comment]
@arfon, do you know what might cause the error above?

**Comment by arfon** [comment]
Looks like an author is missing an affiliation in the `paper.md`:

`  - name: Alberto Vallinotto`

**Comment by eloisabentivegna** [comment]
Thanks!

@amoradinejad, could you add the missing affiliation?

**Comment by amoradinejad** [comment]
The second author does not have an affiliation as he is not in academia. Do
all authors require an affiliation?

Thanks.

**Comment by eloisabentivegna** [comment]
> Do all authors require an affiliation?

They do not. Please refer to the example at https://joss.readthedocs.io/en/latest/submitting.html?highlight=affiliation#example-paper-and-bibliography for how to list authors who do not have an affiliation.

**Comment by amoradinejad** [comment]
@eloisabentivegna <https://github.com/eloisabentivegna> thanks for the info, I updated the paper with the affiliation as described in the link.

**Comment by eloisabentivegna** [comment]
@editorialbot generate pdf

**Comment by amoradinejad** [comment]
The article looks good and I don’t have any correction, apart from the fact that I would like to have my full last name at the citation in the footnote. So please replace “Dizgah et al.” with “Moradinezhad Dizgah et al”. 

Best Regards,
Azadeh Moradinezhad

**Comment by eloisabentivegna** [comment]
@editorialbot check references

**Comment by eloisabentivegna** [comment]
@editorialbot check references

**Comment by eloisabentivegna** [comment]
@amoradinejad, I am not sure how to fix the footer problem you mention, but I am looking into it. In the meantime, I have created a pull request to your repository with some minor (mostly cosmetic) suggestions. Please take a look and integrate, if you approve.

**Comment by eloisabentivegna** [comment]
> I would like to have my full last name at the citation in the footnote

@amoradinejad, there are some instructions about fine-graining author names at https://github.com/openjournals/inara/blob/main/docs/names.md. Could you take a look and try the fix?

**Comment by eloisabentivegna** [comment]
Once you're done fixing the name, could you:
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by eloisabentivegna** [comment]
@editorialbot check references

**Comment by eloisabentivegna** [comment]
@editorialbot generate pdf

**Comment by eloisabentivegna** [comment]
Thanks @amoradinejad. There seems to be an error with the compilation, due to your affiliation this time -- let me look into it.

In the meantime, could you please go through the checklist above? Items 1 and 3 have not been addressed (we're missing the version tag, and the title on Zenodo does not match the submission title).

**Comment by eloisabentivegna** [comment]
@amoradinejad, I believe I have found the source of the problem. Could you accept the pull request, and see if it works?

**Comment by amoradinejad** [comment]
@eloisabentivegna <https://github.com/eloisabentivegna> , sorry for the missing information. 

Here is the checklist:
 Make a tagged release of your software, and list the version tag of the archived version here: Version v2.0.0 <https://zenodo.org/record/6576183>
 Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository): ARCHIVED IN ZENODO
 Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.  CORRECTED THE TITLE ON ZENODO
 Please list the DOI of the archived version here. DOI: 10.5281/zenodo.6576183

**Comment by amoradinejad** [comment]
@eloisabentivegna <https://github.com/eloisabentivegna>, thanks. I accepted the pull request.

**Comment by eloisabentivegna** [comment]
@editorialbot generate pdf

**Comment by eloisabentivegna** [comment]
@amoradinejad, thanks for the further details. But the tag/repository now points to an old version -- could you create a new one?

**Comment by eloisabentivegna** [comment]
@editorialbot check references

**Comment by eloisabentivegna** [comment]
@amoradinejad, the tag refers to the code version before the last few fixes. Can you update it so it points to the current version instead?

**Comment by eloisabentivegna** [comment]
@editorialbot check references

**Comment by eloisabentivegna** [comment]
@amoradinejad, please let me know if you need support with the new tag. Creating a new one should be no different than what you did the first time, but I am happy to guide you step-by-step if need be.

**Comment by amoradinejad** [comment]
@eloisabentivegna <https://github.com/eloisabentivegna>, sorry for a long delay in responding. I created a new tagged release, v2.1.0. Please let me know if now everything looks fine. 


Thanks

**Comment by eloisabentivegna** [comment]
@editorialbot set v2.1.0 as version

**Comment by eloisabentivegna** [comment]
@editorialbot set 10.5281/zenodo.6762607 as archive

**Comment by eloisabentivegna** [comment]
@editorialbot check references

**Comment by eloisabentivegna** [comment]
@editorialbot check pdf

**Comment by eloisabentivegna** [comment]
@editorialbot generate pdf

**Comment by eloisabentivegna** [comment]
Thanks @amoradinejad and co-authors for this nice submission, and @ziotom78 and @abhimaniyar for helping improving it even further. I am satisfied with the result, and am now going to recommend acceptance.

**Comment by eloisabentivegna** [comment]
@editorialbot accept

**Comment by eloisabentivegna** [comment]
@editorialbot recommend-accept

**Comment by danielskatz** [comment]
Also, are bib entries MoradinezhadDizgah:2021dei and oradinezhadDizgah:2021 the same?  If so, can one be removed?

**Comment by danielskatz** [comment]
👋 @amoradinejad - note that publication is almost done, and waiting on some actions from you above...

**Comment by amoradinejad** [comment]
Yes, the two entries are the same. One is now removed.

**Comment by amoradinejad** [comment]
I accepted the merge request. All looks good to me. I updated the journal name to have full names.

**Comment by arfon** [comment]
@editorialbot recommend-accept

**Comment by arfon** [comment]
@editorialbot accept

**Comment by arfon** [comment]
@amoradinejad – could you please merge this PR which fixes an issue with the first author name? https://github.com/amoradinejad/limHaloPT/pull/19

**Comment by amoradinejad** [comment]
Merge request accepted.

**Comment by arfon** [comment]
@editorialbot re-accept

**Comment by arfon** [comment]
@editorialbot reaccept

**Comment by arfon** [comment]
@xuanxu @tarleb – there was an error when publishing this paper (when the author name was written as follows: https://raw.githubusercontent.com/amoradinejad/limHaloPT/262d3fae1c5478e107491b0d462300e7ee107657/Paper/paper.md)

This resulted in the following output on the JOSS site:

<img width="302" alt="Screenshot 2022-07-09 at 22 15 45" src="https://user-images.githubusercontent.com/4483/178122990-0ab0b5d0-1c02-4fcc-a6fa-ad65ff1da18c.png">

I proposed a change which I thought would fix this (see [paper.md](https://raw.githubusercontent.com/amoradinejad/limHaloPT/master/Paper/paper.md)) based on the [Inara docs](
https://github.com/openjournals/inara/blob/main/docs/names.md):

```
authors:
  - given-names: Azadeh 
    surname: Moradinezhad Dizgah
    orcid: 0000-0001-8841-9989
    affiliation: 1
  - name: Alberto Vallinotto  
    affiliation: 2
  - name: Farnik Nikakhtar
    orcid: 0000-0002-3641-4366
    affiliation: 3
```

but it doesn't seem to have worked. Could you advise what the right formatting of the metadata is here?

**Comment by xuanxu** [comment]
I think the error happens in the Ruby code parsing names, currently is not accepting the same keys as Inara. 
I'll fix it and release a new version of the Theoj gem.

**Comment by arfon** [comment]
@editorialbot reaccept
